From: Sam Ravnborg <sam@ravnborg.org>
Date: Sat, 25 Aug 2018 00:16:27 +0200
Subject: [PATCH] skovarm9cpu: add board.mem global variable

board.mem will report the amount of availabe memory in MiB

Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
---
 arch/arm/boards/skovarm9cpu/of_init.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/boards/skovarm9cpu/of_init.c b/arch/arm/boards/skovarm9cpu/of_init.c
index d3320d54b239..fcf460da856b 100644
--- a/arch/arm/boards/skovarm9cpu/of_init.c
+++ b/arch/arm/boards/skovarm9cpu/of_init.c
@@ -13,11 +13,16 @@
  */
 
 #include <common.h>
+#include <globalvar.h>
+#include <magicvar.h>
 #include <envfs.h>
 #include <init.h>
 #include <gpio.h>
 
+#include <linux/sizes.h>
+
 #include <mach/at91sam9263_matrix.h>
+#include <mach/at91sam9_sdramc.h>
 #include <mach/at91sam9_smc.h>
 #include <mach/hardware.h>
 #include <mach/iomux.h>
@@ -90,11 +95,20 @@ static int at91sam9263_smc_init(void)
 }
 device_initcall(at91sam9263_smc_init);
 
+BAREBOX_MAGICVAR_NAMED(board_mem, board.mem,
+        "The detected memory size in MiB");
+
+static int mem;
+
 static int skovarm9cpu_env_init(void)
 {
 	if (!check_fdt())
 		return 1;
 
+	mem = at91_get_sdram_size(IOMEM(AT91SAM9263_BASE_SDRAMC0));
+	mem = mem / SZ_1M;
+	globalvar_add_simple_int("board.mem", &mem, "%u");
+
 	return 0;
 }
 late_initcall(skovarm9cpu_env_init);
